storages/redb-storage/src/core.rs

```diff
@@ -0,0 +1,247 @@
+use {
+    super::error::StorageError,
+    async_stream::try_stream,
+    bincode::{deserialize, serialize},
+    futures::stream::iter,
+    gluesql_core::{
+        data::{Key, Schema},
+        store::{DataRow, RowIter},
+    },
+    redb::{Database, ReadableTable, TableDefinition},
+    std::path::Path,
+    uuid::Uuid,
+};
+
+const SCHEMA_TABLE_NAME: &str = "__SCHEMA__";
+const SCHEMA_TABLE: TableDefinition<&str, Vec<u8>> = TableDefinition::new(SCHEMA_TABLE_NAME);
+
+type Result<T> = std::result::Result<T, StorageError>;
+
+pub struct StorageCore {
+    db: Database,
+    txn: Option<redb::WriteTransaction>,
+    autocommit: bool,
+}
```

Comment:
Does it need `autocommit` if `txn` is `None`?
If not, could we try it like this?
 
```rs
    pub enum TransactionState {
        None,
        Active {
            txn: redb::WriteTransaction,
            autocommit: bool,
        },
    }

    pub struct StorageCore {
        db: Database,
        state: TransactionState,
    }
```

Reply:
wow, thanks a lot! this is certainly better to bind `txn` and `autocommit` to a single enum variant. There is no use case of `autocommit` without `txn`.